install/kubernetes/cilium/charts/agent/templates/daemonset.yaml

```diff
@@ -307,7 +307,7 @@ spec:
           {{- toYaml .Values.initResources | trim | nindent 10 }}
 {{- end }}
       restartPolicy: Always
-{{- if and (eq .Release.Namespace "kube-system") (or (gt .Capabilities.KubeVersion.Minor "10") (gt .Capabilities.KubeVersion.Major "1"))}}
+{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}
```

Comment:
There's a missing `and` in the end, it should be like this:
`{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10"))  (and (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}`

Reply:
This is wrong as well. It must be `or (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1")`. The idea is to match on 2.x, 3.x, 4.x, ... as well.

Reply:
@johngmyers
`{{- if or (and (eq .Release.Namespace "kube-system") (gt .Capabilities.KubeVersion.Minor "10")) (or (gt .Capabilities.KubeVersion.Minor "16") (gt .Capabilities.KubeVersion.Major "1"))}}`

Reply:
"or" is variadic. The problem was an extra close parenthesis.

Reply:
@johngmyers the way it is right now won't work for k8s 2.x, 3.x like Thomas point it out.

Reply:
Why wouldn't it work for k8s 2.x or 3.x? The `(gt .Capabilities.KubeVersion.Major "1")` clause takes care of those cases.

I had only fixed one of the two extraneous close parens. Fixed the other one.

Reply:
sorry I misread the last change. Now seems to be good.